components/client/src/main/java/com/hotels/styx/client/loadbalancing/strategies/PowerOfTwoStrategy.java

```diff
@@ -0,0 +1,140 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.client.loadbalancing.strategies;
+
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.hotels.styx.api.Environment;
+import com.hotels.styx.api.client.ActiveOrigins;
+import com.hotels.styx.api.client.ConnectionPool;
+import com.hotels.styx.api.client.RemoteHost;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategy;
+import com.hotels.styx.api.client.loadbalancing.spi.LoadBalancingStrategyFactory;
+import com.hotels.styx.api.configuration.Configuration;
+
+import java.util.List;
+import java.util.Random;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+
+/**
+ * A Power Of Two load balancing strategy.
+ *
+ * Randomly choose two origins, and return the one with less ongoing requests.
+ */
+public class PowerOfTwoStrategy implements LoadBalancingStrategy {
+    private ActiveOrigins activeOrigins;
+    private Random rng;
+
+    public PowerOfTwoStrategy(ActiveOrigins activeOrigins) {
+        this(activeOrigins, new Random());
+    }
+
+    @VisibleForTesting
+    PowerOfTwoStrategy(ActiveOrigins activeOrigins, Random rng) {
+        this.activeOrigins = activeOrigins;
+        this.rng = rng;
+    }
+
+    /**
+     * A load balancing strategy that favours the origin with the least response time.
+     */
+    public static class Factory implements LoadBalancingStrategyFactory {
+        @Override
+        public LoadBalancingStrategy create(Environment environment, Configuration strategyConfiguration, ActiveOrigins activeOrigins) {
+            return new PowerOfTwoStrategy(activeOrigins);
+        }
+    }
+
+    @Override
+    public Iterable<RemoteHost> vote(Context context) {
+        List<ConnectionPoolStatus> poolsList = stream(activeOrigins.snapshot().spliterator(), false)
+                .map(host -> new ConnectionPoolStatus(host, context))
+                .collect(toList());
+
+        if (poolsList.size() == 0) {
+            return ImmutableList.of();
+        }
+
+        if (poolsList.size() == 1) {
+            return ImmutableList.of(poolsList.get(0).host());
+        }
+
+        ConnectionPoolStatus poolStatus1 = choose(poolsList);
+        ConnectionPoolStatus poolStatus2 = choose(poolsList, poolStatus1);
+
+        return betterOf(poolStatus1, poolStatus2);
+    }
+
+    private Iterable<RemoteHost> betterOf(ConnectionPoolStatus poolStatus1, ConnectionPoolStatus poolStatus2) {
```

Comment:
The function to decide which pool is better could be decoupled and passed in - since the concept of choosing the better of two randoms is independent of what you consider "better" to mean.